libs/components/inline-form/src/lib/modules/inline-form/inline-form.component.ts

```diff
@@ -132,152 +137,130 @@ export class SkyInlineFormComponent implements OnInit, OnDestroy {
-      this.buttons = buttons;
-      this.#changeDetectorRef.markForCheck();
-    });
+    this.#getPresetButtons()
+      .pipe(takeUntil(this.#ngUnsubscribe))
+      .subscribe((buttons: SkyInlineFormButtonConfig[]) => {
+        this.buttons = buttons;
+        this.#changeDetectorRef.markForCheck();
+      });
   }
 
-  private getPresetButtons(): Promise<SkyInlineFormButtonConfig[]> | undefined {
+  #getPresetButtons(): Observable<SkyInlineFormButtonConfig[]> {
+    const emitter = new ReplaySubject<SkyInlineFormButtonConfig[]>(1);
+
     const buttonType = this.config
       ? this.config.buttonLayout
       : SkyInlineFormButtonLayout.DoneCancel;
 
-    let promise: Promise<SkyInlineFormButtonConfig[]> | undefined;
-
     switch (buttonType) {
       /* istanbul ignore next */
       default:
       case SkyInlineFormButtonLayout.DoneCancel:
         observableZip(
           this.#resourcesService.getString('skyux_inline_form_button_done'),
           this.#resourcesService.getString('skyux_inline_form_button_cancel')
-        )
-          .pipe(take(1))
-          .subscribe((values: string[]) => {
-            promise = new Promise<SkyInlineFormButtonConfig[]>(
-              (resolve: (value: SkyInlineFormButtonConfig[]) => void) => {
-                resolve([
-                  {
-                    text: values[0],
-                    styleType: 'primary',
-                    action: 'done',
-                  },
-                  {
-                    text: values[1],
-                    styleType: 'link',
-                    action: 'cancel',
-                  },
-                ]);
-              }
-            );
-          });
+        ).subscribe((values: string[]) => {
+          emitter.next([
+            {
+              text: values[0],
+              styleType: 'primary',
+              action: 'done',
+            },
+            {
+              text: values[1],
+              styleType: 'link',
+              action: 'cancel',
+            },
+          ]);
+        });
         break;
 
       case SkyInlineFormButtonLayout.SaveCancel:
         observableZip(
           this.#resourcesService.getString('skyux_inline_form_button_save'),
           this.#resourcesService.getString('skyux_inline_form_button_cancel')
-        )
-          .pipe(take(1))
-          .subscribe((values: string[]) => {
-            promise = new Promise<SkyInlineFormButtonConfig[]>(
-              (resolve: (value: SkyInlineFormButtonConfig[]) => void) => {
-                resolve([
-                  {
-                    text: values[0],
-                    styleType: 'primary',
-                    action: 'save',
-                  },
-                  {
-                    text: values[1],
-                    styleType: 'link',
-                    action: 'cancel',
-                  },
-                ]);
-              }
-            );
-          });
+        ).subscribe((values: string[]) => {
```

Comment:
Does this subscription need a `takeUntil(this.#ngUnsubscribe)`?

Reply:
done in: [pipe takeUntil unsubscribe on resource string subscriptions](https://github.com/blackbaud/skyux/pull/670/commits/d720487ef67be7bf88df3aba4ae495d50b5ee070)